inc/commondevice.class.php

```diff
@@ -407,27 +407,44 @@ function import(array $input) {
             $compare = explode(':', $compare);
             switch ($compare[0]) {
                case 'equal':
-                  $where[] = "`".$field."`='".$input[$field]."'";
+                  $where[$field] = $input[$field];
                   break;
 
                case 'delta':
-                  $where[] = "`".$field."`>'".((int) $input[$field] - (int) $compare[1])."'";
-                  $where[] = "`".$field."`<'".((int) $input[$field] + (int) $compare[1])."'";
+                  $where[] = [
+                     [$field => ['>', ((int) $input[$field] - (int) $compare[1])]],
+                     [$field => ['<', ((int) $input[$field] + (int) $compare[1])]]
+                  ];
                   break;
             }
          }
       }
 
-      $query = "SELECT `id`
-                FROM `".$this->getTable()."`
-                WHERE ".  implode(" AND ", $where);
+      $iterator = $DB->request([
+         'SELECT' => ['id'],
+         'FROM'   => $this->getTable(),
+         'WHERE'  => $where
+      ]);
 
-      $result = $DB->query($query);
-      if ($DB->numrows($result) > 0) {
-         $line = $DB->fetch_assoc($result);
+      if (count($iterator) == 1) {
```

Comment:
I am not sure that adding this check is a good idea. If there are multiple values existing, you will now return `false` instead of a valid `id`. Maybe it will result in bugs on following usages of this function :
 - https://github.com/glpi-project/glpi/blob/9.2/bugfixes/inc/transfer.class.php#L1205
 - https://github.com/glpi-project/glpi/blob/9.2/bugfixes/inc/transfer.class.php#L2515
 - https://github.com/glpi-project/glpi/blob/9.2/bugfixes/inc/transfer.class.php#L2558

Reply:
I really do not know what is expected... Anyways, this did not exists in current code, so I'll drop it.